### PR TITLE
Add CSYNC record from RFC7477

### PR DESCRIPTION
**Pull request by frelon**
The CSYNC record is used to synchronize records (mostly NS and glue) to the parent zone.

Had to move nsec3::encode_type_bit_maps and decode_type_bit_maps in order to make it work with/without the dnssec-feature.

Tested with the following zone: 

```
@   IN          SOA     trust-dns.org. root.trust-dns.org. (
                                199609203       ; Serial
                                28800   ; Refresh
                                7200    ; Retry
                                604800  ; Expire
                                86400)  ; Minimum TTL

                CSYNC   66  3   A NS MX
```
And get the following output using resolve util:

```
$ ./target/release/resolve -n 127.0.0.1:24141 -t CSYNC example.com
Querying for example.com CSYNC from tcp:127.0.0.1:24141, udp:127.0.0.1:24141
Success for query name: example.com type: CSYNC class: IN mdns_unicast_response: false
        example.com. 86400 IN CSYNC 66 3 A NS MX
```

### CODE REVIEW

File: crates/client/src/serialize/txt/rdata_parsers/csync.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    Ok(CSYNC::new(soa_serial, immediate, soa_minimum, record_types))
+}
+
+#[test]
+fn test_parsing() {
+    // IN HINFO DEC-2060 TOPS20
+
+    assert_eq!(
+        parse(vec!["DEC-2060", "TOPS20"].into_iter()).expect("failed to parse NAPTR"),
```
Comment (bluejekyll) [inline crates/client/src/serialize/txt/rdata_parsers/csync.rs:52]: Are these tests correct?

File: crates/proto/src/rr/record_data.rs [hunk 2]
```diff
@@ -167,6 +167,10 @@ pub enum RData {
     /// ```
     CNAME(Name),
 
+    /// ```text
```
Comment (bluejekyll) [inline crates/proto/src/rr/record_data.rs:170]: Can you add the text that talks about it's use in DNS?

**Review by bluejekyll — COMMENTED** [review]
A couple of comments, but this looks really good. Thanks for the PR! And I agree with the move of the type_bit_map function.

File: crates/client/src/serialize/txt/rdata_parsers/csync.rs (hunk 1 above)
Comment (frelon) [reply crates/client/src/serialize/txt/rdata_parsers/csync.rs]: Hehe I was a bit too fast on the copy! Fixed now! :+1:

File: crates/proto/src/rr/record_data.rs (hunk 2 above)
Comment (frelon) [reply crates/proto/src/rr/record_data.rs:170]: Absolutely, copied in some text from the RFC!

**Review by bluejekyll — APPROVED** [review]
This all looks good to me. Thanks for the PR!

**Comment by frelon** [comment]
> This all looks good to me. Thanks for the PR!

Great! It took some time with the formatting and testing but now it looks green :+1:

**Pull request merged**